### PR TITLE
Coverage

### PR DESCRIPTION
**Pull request by MHBauer**
not sure this would be worth it, other than the makefile cleanup.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
-sudo: required
-language: bash
-
-services:
-  - docker
+language: go
```
Comment (duglin) [inline .travis.yml]: Do we need this for codecov? One of the reasons I didn't want to tag this as a "go" project is because at some point I'm hoping to have the CF ruby code move into here too - which means we're not just go.

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
 
 script:
   - make all
+  
+after_success:
+  - bash generate-coverage.sh
```
Comment (duglin) [inline .travis.yml]: Could this be part of the lower-level Makefile so its go-specific?

File: k8s/service_controller/Makefile [hunk 3]
```diff
@@ -14,10 +14,10 @@ binary: $(EXE)
 $(EXE): $(SRC_FILES)
 ifeq (,$(shell which go))
 	@$(MAKE) --no-print-directory binary-docker
+	docker build -t $(IMAGE):latest .
 else
 	@$(MAKE) --no-print-directory binary-local
 endif
-	docker build -t $(IMAGE):latest .
```
Comment (duglin) [inline k8s/service_controller/Makefile]: Why did you move this up?

Comment (duglin) [inline k8s/service_controller/Makefile]: Even on a local build I still want to build the image since at some point I expect it to be run via docker in the tests.

File: k8s/service_controller/brokers/go/Makefile [hunk 4]
```diff
@@ -5,10 +5,10 @@ all: $(EXE)
 $(EXE): gobroker.go
 ifeq (,$(shell which go))
 	@$(MAKE) --no-print-directory binary-docker
+	docker build -t gobroker .
 else
 	@$(MAKE) --no-print-directory binary-local
 endif
-	docker build -t gobroker .
```
Comment (duglin) [inline k8s/service_controller/brokers/go/Makefile]: Ditto here - irrespective of whether we build with docker or not we should still create an image.

File: .travis.yml (hunk 1 above)
Comment (MHBauer) [inline .travis.yml]: can try using `gimme` manually, but travis is not set up to do multi language projects.

File: .travis.yml (hunk 2 above)
Comment (MHBauer) [inline .travis.yml]: I don't understand. It is a script.

File: k8s/service_controller/Makefile (hunk 3 above)
Comment (MHBauer) [inline k8s/service_controller/Makefile]: because I don't want to need docker in travis.

Comment (duglin) [inline k8s/service_controller/Makefile]: we won't be able to avoid it at some point since we'll need to run kube for our tests and right now I think that means hyperkube in a container(s).  Embrace the inevitable  :-)

File: .travis.yml (hunk 2 above)
Comment (duglin) [inline .travis.yml]: But let's run it outside of the travis stuff since the code coverage will be go, or ruby, specific depending on which sub-project we're building/testing.  Moving it into the k8s portion will allow us to run the go version for just the k8s side of things.

File: .travis.yml (hunk 1 above)
Comment (duglin) [inline .travis.yml]: the current setup is setup to do multi languages - its doing go w/o travis knowing it.  So we should be able to add ruby for the CF side of the project as well.
